core/insertion_marker_manager.ts

```diff
@@ -561,8 +555,31 @@ export class InsertionMarkerManager {
       // probably recreate the marker block (e.g. in getCandidate_), which is
       // called more often during the drag, but creating a block that often
       // might be too slow, so we only do it if necessary.
-      this.firstMarker = this.createMarkerBlock(this.topBlock);
-      insertionMarker = isLastInStack ? this.lastMarker : this.firstMarker;
+      if (isLastInStack && this.lastOnStack) {
+        if (this.lastMarker) {
+          eventUtils.disable();
+          try {
+            this.lastMarker.dispose();
+          } finally {
+            eventUtils.enable();
+          }
+        }
+        this.lastMarker =
+            this.createMarkerBlock(this.lastOnStack.getSourceBlock());
+        insertionMarker = this.lastMarker;
+      } else {
+        if (this.firstMarker) {
+          eventUtils.disable();
+          try {
+            this.firstMarker.dispose();
+          } finally {
+            eventUtils.enable();
+          }
+        }
```

Comment:
Oof, were we just eating memory before?

Reply:
I assume the GC would have cleaned up after us eventually since the marker would have become unreachable, but you introduced this logic in #4184 to fix an issue so I'm carrying it forward.